rpm-ostree-bisect

```diff
@@ -0,0 +1,347 @@
+        # Verify test script exists and is executable
+        if not testscript:
+            fatal("Must provide a --testscript to run")
+        if not (os.path.isfile(testscript)
+             and os.access(testscript, os.X_OK)):
+            fatal("provided test script: %s is not an executable file"
+                    % testscript)
+
+        # Assume currently booted commit is bad if no
+        # bad commit was given
+        if badcommit is None:
+            badcommit = deployment.get_csum()
+
+        # pull commit history
+        pull_commit_history(deployment, repo)
+
+        # initialize data
+        commits_info = initialize_commits_info(repo, 
+                                               badcommit, 
+                                               goodcommit)
+
+        # Write data to file
+        write_data(datafile, commits_info, testscript)
+
+        # write/enable systemd unit
+        with open(SYSTEMD_UNIT_FILE, 'w') as f:
+            f.write(SYSTEMD_UNIT_CONTENTS)
+        cmd = ['/usr/bin/systemctl', 'daemon-reload']
+        subprocess.run(cmd)
+        cmd = ['/usr/bin/systemctl', 'enable', SYSTEMD_UNIT_NAME]
+        subprocess.run(cmd)
+        return 0
+
+    else:
+        # load data
+        commits_info, testscript = load_data(datafile)
+
+        # run test
+        completed = subprocess.run(testscript, shell=True, check=False)
+        if completed.returncode == 0:
+            success = True
+        else:
+            success = False
+
+        # update and write data
+        commit = deployment.get_csum()
+        if success:
+            for c in reversed(commits_info.keys()):
+                commits_info[c]['status'] = 'GOOD' 
+                if c == commit:
+                    commits_info[c]['heuristic'] = 'TESTED' 
+                    break
+        else:
+            for c in commits_info.keys():
+                commits_info[c]['status'] = 'BAD' 
+                if c == commit:
+                    commits_info[c]['heuristic'] = 'TESTED' 
+                    break
+        write_data(datafile, commits_info, testscript)
+
+        # Find list of unknown status commits
+        unknowns = []
+        lastbad = None
+        firstgood = None
+        for commitid in commits_info.keys():
+            status = commits_info[commitid]['status'] 
+            if status == 'BAD':
+                lastbad = commitid
+            elif status == 'UNKNOWN':
+                unknowns.append(commitid)
+            elif status == 'GOOD':
+                firstgood = commitid
+                break
+
+        if len(unknowns) == 0:
+            # We're done!
+            log("Last known good commit:\n  %s : %s" %
+                  (firstgood, commits_info[firstgood]['version']))
+            log("First known bad commit:\n  %s : %s" %
+                  (lastbad, commits_info[lastbad]['version']))
+            # print out some db info
+            pull_commit_history(deployment, repo)
```

Comment:
Hmm, why do we need to do this again?

Reply:
because when we 'deploy' we lose that commit meta data (it gets cleaned up). We want to re-pull the commit metadata so we can do the diff below.